Sources/NIOPosix/Bootstrap.swift

```diff
@@ -1352,6 +1352,482 @@ public final class ClientBootstrap: NIOClientTCPBootstrapProtocol {
     }
 }
 
+// MARK: AsyncChannel based connect
+extension ClientBootstrap {
+    /// Specify the `host` and `port` to connect to for the TCP `Channel` that will be established.
+    ///
+    /// - Parameters:
+    ///   - host: The host to connect to.
+    ///   - port: The port to connect to.
+    ///   - channelInboundType: The channel's inbound type.
+    ///   - channelOutboundType: The channel's outbound type.
+    ///   - channelBackpressureStrategy:  The back pressure strategy used by the channel.
+    ///   - isOutboundHalfClosureEnabled:  Indicates if half closure is enabled on the channel. If half closure is enabled
+    ///   then finishing the `NIOAsyncChannelWriter` will lead to half closure.
+    /// - Returns: A `NIOAsyncChannel` for the established connection.
+    @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+    @_spi(AsyncChannel)
+    public func connect<ChildChannelInbound: Sendable, ChildChannelOutbound: Sendable>(
+        host: String,
+        port: Int,
+        channelInboundType: ChildChannelInbound.Type = ChildChannelInbound.self,
+        channelOutboundType: ChildChannelOutbound.Type = ChildChannelOutbound.self,
```

Comment:
I think these type names are wrong: there are no child channels here. Same for the other `connect` functions added in this PR.